core/dbt/events/types.proto

```diff
@@ -1585,6 +1586,85 @@ message RunResultWarningMessage {
     string msg = 2;
 }
 
+// Z047
+message DebugEnvironmentDetails {
```

Comment:
So I added a lot of these.

1. Jerco considered differentiating between [Successes and Failures](https://github.com/dbt-labs/dbt-core/issues/5353#issuecomment-1152299423). I erred on that side.

2. I tacked these onto Q types, but Node Execution is sort of a bit loose (may even be erroneous). My considered alternative was to split this off into, like, a "Task Execution" subcategory 🤷‍♀️ Figured I'd check with reviewers.